tests/rustdoc-json/doc_hidden_reexports.rs

```diff
@@ -0,0 +1,19 @@
+// Regression test for <https://github.com/rust-lang/rust/issues/117718>.
+// Ensures that the various "doc(hidden)" reexport cases are correctly handled.
+
+pub mod submodule {
+    #[doc(hidden)]
+    pub struct Hidden {}
+}
+
+#[doc(hidden)]
+mod x {
+    pub struct Y {}
+}
+
+// @has "$.index[*].inner[?(@.import.name=='UsedHidden')].import.source" '"submodule::Hidden"'
+// @has "$.index[*].inner[?(@.import.name=='UsedHidden')].import.id" "null"
+pub use submodule::Hidden as UsedHidden;
+// @has "$.index[*].inner[?(@.import.name=='Z')].import.source" '"x::Y"'
+// @has "$.index[*].inner[?(@.import.name=='Z')].import.id" 'null'
```

Comment:
As discussed in <https://rust-lang.zulipchat.com/#narrow/stream/266220-t-rustdoc/topic/pub.20re-export.20of.20item.20from.20.60doc.28hidden.29.60.20module/near/401627083>, this import should be a part of the output, and so the ID should be present

Reply:
> I followed the same logic as for reexport of item inside private module.

When reexporting an item defined in a private module, the import will have the ID of the underlying item. We should do the same thing here.

> We discussed that after this PR, we want to inline such items

Please no. Inlining was a massive source of bugs for rustdoc-json, and removing it was a great improvement. I'm strongly opposed to bringing it back.

Reply:
Having an ID pointing to an item that is not available means that the current bug fixed by this PR will remain.